opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/point.py

```diff
@@ -184,6 +187,11 @@ class ResourceMetrics:
     scope_metrics: Sequence[ScopeMetrics]
     schema_url: str
 
+    def __len__(self):
+        return sum(
+            [len(scope_metrics) for scope_metrics in self.scope_metrics]
```

Comment:
I find it odd that scope metrics are summed here. I would imagine this returns just the len of scope_metrics.

Reply:
I need the number of metrics in the whole hierarchy...

Reply:
These `__len__` methods are misleading and should not be added to `ResourceMetrics` nor to `ScopeMetrics`. None of these classes should be treated as a sequence, they are not. What are sequences are their `scope_metrics` and `metrics` attributes respectively.